src/utils/self_test/self_test.c

```diff
@@ -434,72 +693,1012 @@ static void print_results(struct st_latency *latencies,
 	qsort(latencies, test_params->rep_count,
 	      sizeof(latencies[0]), st_compare_latencies_by_ranks);
 
-	printf("\tEndpoint results (rank:tag - Median Latency (us)):\n");
+#ifdef PRINT_ALL
+	printf("\n NEW Sorted value\n");
+	for (local_rep = num_failed; local_rep < test_params->rep_count;
+	     local_rep++){
+		printf("   Iteration: %d, EP %d, latencies %ld\n",
+		       local_rep,  latencies[local_rep].rank,
+		       latencies[local_rep].val);
+	}
+	printf("\n");
+#endif
 
 	/* Iterate over each rank / tag pair */
 	local_rep = 0;
 	do {
 		uint32_t rank = latencies[local_rep].rank;
 		uint32_t tag = latencies[local_rep].tag;
-		uint32_t start_idx = local_rep;
-		uint32_t last_idx;
-		uint32_t median_idx;
+		uint32_t num_used;
+		uint32_t count;
+		uint32_t begin;
 
 		/* Compute start, last, and num_failed for this rank/tag */
 		num_failed = 0;
+		count = 0;
+		begin = local_rep;
 		while (1) {
 			if (latencies[local_rep].rank != rank ||
 			    latencies[local_rep].tag != tag)
 				break;
 
-			if (latencies[local_rep].cci_rc < 0)
-				num_failed++;
-
+			count++;
 			local_rep++;
 			if (local_rep >= test_params->rep_count)
 				break;
 		}
-		last_idx = local_rep - 1;
-		D_ASSERT(start_idx + num_failed <= last_idx);
+		D_ASSERT(count > 0);
+		/* Find stats for this endpoint */
+		num_used = calculate_stats(&latencies[begin],
+					   count,
+					   &latency_avg, &latency_std_dev,
+					   &latency_min, &latency_max,
+					   &latency_med25, &latency_med50,
+					   &latency_med75, &latency_total);
+		D_ASSERT(num_used > 0);
+
+		num_failed = count - num_used;
+		if (num_failed > 0) {
+			printf("\n");
+			printf("\t\t\tFailures: %u\n", num_failed);
+			print_fail_counts(&latencies[begin], num_failed,
+					  "\t\t\t");
+		}
 
-		/* Compute median index */
-		median_idx = start_idx + num_failed +
-			(last_idx - start_idx - num_failed) / 2;
-		D_ASSERT(median_idx <= last_idx);
-		D_ASSERT(median_idx >= start_idx + num_failed);
+		/* 
+ 		 * Throughput and bandwidth calculations.
+ 		 * scale = 1000000000;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs